shared/utils/error_classifier.py

```
@@ -0,0 +1,361 @@
+
+# Map known SDK exception class names to error codes.
+# Uses class name strings to avoid hard dependencies on SDK packages.
+_EXCEPTION_CLASS_MAP: dict[str, ChatErrorCode] = {
+    # OpenAI SDK
+    "RateLimitError": ChatErrorCode.RATE_LIMIT,
+    "AuthenticationError": ChatErrorCode.FORBIDDEN,
+    "PermissionDeniedError": ChatErrorCode.FORBIDDEN,
+    "NotFoundError": ChatErrorCode.MODEL_UNAVAILABLE,
+    "InternalServerError": ChatErrorCode.MODEL_UNAVAILABLE,
+    # Anthropic SDK
+    "RateLimitError": ChatErrorCode.RATE_LIMIT,
```

Ruff (0.15.9)

[error] Dictionary key literal `"RateLimitError"` repeated

Remove repeated key literal `"RateLimitError"`

(F601)